element-connector/src/main/java/org/eclipse/californium/elements/RelaxedCorrelationContextMatcher.java

```diff
@@ -47,10 +49,10 @@ private final boolean internalMatch(CorrelationContext requestedContext, Correla
 		} else if (null == availableContext) {
 			return false;
 		}
-		if (requestedContext.get(DtlsCorrelationContext.KEY_SESSION_ID) != null) {
-			return CorrelationContextUtil.match(getName(), KEYS, requestedContext, availableContext);
-		}
-		return requestedContext.equals(availableContext);
+		Set<String> keys = new CopyOnWriteArraySet<String>(requestedContext.getMatchingKeys());
+		keys.addAll(availableContext.getMatchingKeys());
```

Comment:
I prefer to explicitly set the key used for matching. This is less error prone in case where somebody add new elements in the context.